src/main/kotlin/frc/robot/subsystems/Visualizer.kt

```diff
@@ -1,97 +1,156 @@
+import frc.robot.subsystems.drive.Drive
 import org.littletonrobotics.junction.AutoLogOutput
 
+private val swerveMoudlePose: Array<Translation2d> =
+    Drive.getModuleTranslations()
+
 private val INITIAL_INTAKE_TRANSLATION =
-    getTranslation3d(x = Meters.of(-0.25), z = Meters.of(0.35))
+    getTranslation3d(x = Meters.of(-0.32), z = Meters.of(0.35))
+private val INITIAL_INTAKE_Roller_TRANSLATION =
+    getTranslation3d(x = Meters.of(-0.32 + 0.62890), z = Meters.of(0.35))
 
 private val INITIAL_WRIST_TRANSLATION =
-    getTranslation3d(x = Meters.of(0.08), z = Meters.of(0.43))
+    getTranslation3d(x = Meters.of(0.27434), z = Meters.of(0.79707))
+
+private val INITIAL_ALGEA_REMOVER_ROLLER_TRANSLATION =
+    getTranslation3d(x = Meters.of(0.113), z = Meters.of(0.995))
+
+private val INITIAL_Elevator_1_TRANSLATION =
+    getTranslation3d(x = Meters.of(0.11250), z = Meters.of(0.14345040))
+
+private val INITIAL_Elevator_2_TRANSLATION =
+    getTranslation3d(x = Meters.of(0.11250), z = Meters.of(0.20545040))
 
 private val INITIAL_CLIMBER_TRANSLATION =
-    getTranslation3d(x = Meters.of(-0.24), z = Meters.of(0.27))
-
-private val INITIAL_ELEVATOR_TRANSLATION = getTranslation3d(x = 0.12)
-
-class Visualizer(
-    private val extenderPosition: () -> Distance,
-    private val intakeRollerAngle: () -> Angle,
-    private val elevatorHeight: () -> Distance,
-    private val wristAngle: () -> Angle,
-    private val coralRollersAngle: () -> Angle,
-    private val algaeRemoverAngle: () -> Angle,
-    private val climberAngle: () -> Angle,
-) {
+    getTranslation3d(x = Meters.of(-0.24), z = Meters.of(0.285))
+
+class Visualizer {
+    private val swerveDrive = frc.robot.swerveDrive
+    private val climber = frc.robot.climber
+    private val elevator = frc.robot.elevator
+    private val gripper = frc.robot.gripper
+    private val extender = frc.robot.extender
+    private val roller = frc.robot.roller
+    private val wrist = frc.robot.wrist
     private fun getElevatorPoses(): Pair<Pose3d, Pose3d> {
-        val secondStageHeight = elevatorHeight.invoke().`in`(Meters)
+        val secondStageHeight = elevator.height.invoke().`in`(Meters)
         val firstStageHeight = secondStageHeight / 2.0
 
         val firstStagePose =
             getPose3d(
-                INITIAL_ELEVATOR_TRANSLATION +
-                    getTranslation3d(z = firstStageHeight)
+                x = INITIAL_Elevator_1_TRANSLATION.x,
+                z = firstStageHeight.plus(INITIAL_Elevator_1_TRANSLATION.z)
             )
         val secondStagePose =
             getPose3d(
-                INITIAL_ELEVATOR_TRANSLATION +
-                    getTranslation3d(z = secondStageHeight)
+                x = INITIAL_Elevator_2_TRANSLATION.x,
+                z = secondStageHeight.plus(INITIAL_Elevator_2_TRANSLATION.z)
             )
         return Pair(firstStagePose, secondStagePose)
     }
 
+    private fun getSwerveModulePose(): Pair<Array<Pose3d>, Array<Pose3d>> {
```

Comment:
It's better to process a single module, and then have the for loop somewhere else (in another function if you prefer).